sympy/printing/maple.py

```diff
@@ -0,0 +1,172 @@
+"""
+Maple code printer
+
+The MapleCodePrinter converts single sympy expressions into single
+Maple expressions, using the functions defined in the Maple objects where possible.
+
+
+FIXME: This module is still under actively developed. Some functions may be not completed.
+"""
+
+from __future__ import print_function, division
+
+from sympy.codegen.ast import Assignment
+from sympy.core import S
+from sympy.core.compatibility import string_types, range
+from sympy.printing.codeprinter import CodePrinter
+from sympy.printing.precedence import precedence, PRECEDENCE
+
+_known_func_same_name = [
+    'sin',
+    'cos',
+    'tan',
+    'sec',
+    'csc',
+    'cot',
+
+    'sinh',
+    'cosh',
+    'tanh',
+    'sech',
+    'csch',
+    'coth',
+
+    'exp',
+
+    'floor'
+
+]
+
+known_functions = {
+    # Sympy -> Maple
+    # FIXME: maybe need added
+    'Abs': 'abs',
+    'log': 'ln',
+
+    'asin': 'arcsin',
+    'acos': 'arccos',
+    'atan': 'arctan',
+    'asec': 'arcsec',
+    'acsc': 'arccsc',
+    'acot': 'arccot',
+
+    'asinh': 'arcsinh',
+    'acosh': 'arccosh',
+    'atanh': 'arctanh',
+    'asech': 'arcsech',
+    'acsch': 'arccsch',
+    'acoth': 'arccoth',
+
+    'ceiling': 'ceil',
+
+}
+
+for _func in _known_func_same_name:
+    known_functions[_func] = _func
+
+
+class MapleCodePrinter(CodePrinter):
+    """
+    Printer which converts single sympy expressions into single.
+    """
+
+    def __init__(self, settings=None):
+        if settings is None:
+            settings = dict()
+        CodePrinter.__init__(settings)
+        self.known_functions = dict(known_functions)
+        userfuncs = settings.get('user_functions', {})
+        self.known_functions.update(userfuncs)
+
+    def _get_statement(self, codestring):
+        return "%s;" % codestring
+
+    def _get_comment(self, text):
+        return "# {0}".format(text)
+
+    def _declare_number_const(self, name, value):
+        return "{0} := {1};".format(name, value.evalf(self._settings['precision']))
+
+    def _print_Pow(self, expr):
+        PREC = precedence(expr)
+        if expr.exp == -1:
+            return '1/%s' % (self.parenthesize(expr.base, PREC))
+        elif expr.exp == 0.5 or expr.exp == S(1) / 2:
+            return 'sqrt(%s)' % self._print(expr.base)
+        else:
+            return '%s^%s' % (self._print(expr.base),
+                              self._print(expr.exp))
+
+    def _print_Rational(self, expr):
+        p, q = int(expr.p), int(expr.q)
+        return '%d/%d' % (p, q)
+
+    def _print_Relational(self, expr):
+        lhs_code = self._print(expr.lhs)
+        rhs_code = self._print(expr.rhs)
+        op = expr.rel_op
+        # FIXME: rel_op might be different from maple.
+        return "{0} {1} {2}".format(lhs_code, op, rhs_code)
+
+    def _print_Idx(self, expr):
+        return self._print(expr.label)
+
+    def _print_Exp1(self, expr):
+        return "exp(1)"
+
+    def _print_Pi(self, expr):
```

Comment:
It may, in the long run and maybe even from the beginning, make sense to write a function `_print_AtomicExpr` that uses a dict to translate names similar to the planned `_print_Function`. This can be used for Pi, Exp1, Infinity and so on. The current approach is probably slightly faster, but it is easier to add constants in the `_print_AtomicExpr`. A possible drawback is that there may be more than "constants" that are `AtomicExpr`, but one can always print the class-name as a fallback...

( Not sure why no(?) other printer does it this way, now that I write about it...)

Reply:
Yes, it is a very good idea.